agent/src/beerocks/slave/son_slave_thread.cpp

```diff
@@ -515,37 +554,8 @@ bool slave_thread::handle_cmdu_control_message(Socket *sd,
             return false;
         }
 
-        std::string client_mac = network_utils::mac_to_string(request_in->params().mac);
-        std::string client_bridge_4addr_mac =
-            network_utils::mac_to_string(request_in->params().bridge_4addr_mac);
-        std::string client_ip = network_utils::ipv4_to_string(request_in->params().ipv4);
-
-        LOG(DEBUG) << "START_MONITORING_REQUEST: mac=" << client_mac << " ip=" << client_ip
-                   << " bridge_4addr_mac=" << client_bridge_4addr_mac;
-
-        if (request_in->params().is_ire) {
-            auto request_out = message_com::create_vs_message<
-                beerocks_message::cACTION_APMANAGER_CLIENT_IRE_CONNECTED_NOTIFICATION>(cmdu_tx);
-            if (request_out == nullptr) {
-                LOG(ERROR) << "Failed building ACTION_APMANAGER_CLIENT_IRE_CONNECTED_NOTIFICATION "
-                              "message!";
-                return false;
-            }
-            request_out->mac() = request_in->params().mac;
-            message_com::send_cmdu(ap_manager_socket, cmdu_tx);
-        }
-
-        //notify monitor
-        auto request_out = message_com::create_vs_message<
-            beerocks_message::cACTION_MONITOR_CLIENT_START_MONITORING_REQUEST>(
-            cmdu_tx, beerocks_header->id());
-        if (request_out == nullptr) {
-            LOG(ERROR) << "Failed building ACTION_MONITOR_CLIENT_START_MONITORING_REQUEST message!";
-            return false;
-        }
-        request_out->params() = request_in->params();
-        message_com::send_cmdu(monitor_socket, cmdu_tx);
-        break;
+        LOG(DEBUG) << "Call start_monitoring from ACTION_CONTROL_CLIENT_START_MONITORING_REQUEST";
+        return start_monitoring(request_in->params(), beerocks_header);
```

Comment:
What about stop_monitoring? Should be a separate API for that, and it should be called when a client disassociates, doesn't it?